lib/scan_beacon/generic_individual_advertiser.rb

```diff
@@ -0,0 +1,28 @@
+module ScanBeacon
+  class GenericIndividualAdvertiser < GenericAdvertiser
+    def initialize(opts={})
+      @advertising = false
+      @parser = nil
+      super(opts)
+    end
+
+    def beacon=(value)
+      @beacon = value
+      update_ad
+    end
+
+    def parser=(value)
+      @parser = value
+      update_ad
+    end
+
+    def update_ad    
+      self.ad = @parser.generate_ad(@beacon) if @parser && @beacon
+      self.start if @advertising
+    end
+    
+    def ad=(value)
+      @ad = value
+    end
+  end
+end
```

Comment:
I started with that, but then the code for these methods was duplicated in each of the  `GenericIndividualAdvertiser` implementations.   I don't really like the three-tier structure, but I though it was better than the combination of the duplicated code and the fact that we really don't want those methods on all subtypes of `GenericAdvertiser`